### DISCUSSION TITLE
File locations - within FCP library?

### DISCUSSION THREAD

**Discussion by thermidorthelobster** [latenitefilms/BRAWToolbox#82] [Feature Requests]
So sooner or later I'm going to mess up my media management and end up deleting BRAW files and breaking something, because I'm one of those people who likes that FCP does it all for me.

Is it possible that BRAW Toolbox could copy the original BRAW media (and the audio too) into a folder within the FCPX fcpbundle itself? That way the media always goes with the library. Having stuff outside the library seems to be a recipe for me to screw something up :)

I note that I can already create a folder for the audio inside an fcpbundle and keep the audio there; I can't select a folder inside an fcpbundle in BRAW Toolbox, but I can drag a folder from a Finder window into BRAW Toolbox's selection window and it cheerfully accepts the path.

I can do a similar thing if I copy BRAW media into a folder I've created in the fcpbundle before import, then drag that folder from the Finder into BRAW Toolbox's Import BRAW window; but it would be great if there was an option to select the relevant library from within BRAW Toolbox and for it to create the folders and copy the media directly into the library.

**Comment by latenitefilms** [comment]
This is a super interesting idea, but I feel like it's a tricky one for us to implement, because the user might have a LOT of BRAW files which will take a long time to copy. It also brings with it a lot of complexity - what if a users aborts the copy, what if there's not enough space on the destination drive, etc. I think for now it's probably better/easier if you manage these files manually.

One tiny improvement that might help is that we could show library bundles as normal folders in the **Import BRAW Files** window, so you can easily "go inside" the library to select things?

**Comment by thermidorthelobster** [comment]
Oh, yes please - that’s definitely worth doing. Thanks!

Message ID: <latenitefilms/BRAWToolbox/repo-discussions/82/comments/4962241@
github.com>

**Reply by latenitefilms** [reply]
Added a feature request here: https://github.com/latenitefilms/BRAWToolbox/issues/83

**Comment by thermidorthelobster** [comment]
One thought is that this could be a two-stage process - such as Prepare,
where BRAW Toolbox would copy the BRAW files into a folder within the
library you pointed it at, restricting the user from doing anything else
until everything had copied across, then the Import BRAW Files part of the
process could automatically look at the folder in the library it’d just
created. That way any problems with the Prepare part (such as missing files
or full drives) would be self-evident in the second stage (as the files
would be missing) without the need for any particular error-handling?

But I’m happy with folders in library bundles for now :)

Message ID: <latenitefilms/BRAWToolbox/repo-discussions/82/comments/4962241@
github.com>

**Reply by latenitefilms** [reply]
Although it sounds simple, the act of simply copying files from one place to the other is actually pretty complex - hence the reason tools like Hedge, Shotput Pro, Silverstack, YoYotta, etc exist. I think you'd be better off using something like [Hedge](https://hedge.video/hedge) to copy all your camera rushes directly into your library before you start cutting? That way you can ensure you have everything safely copied across, with checksums. 

Final Cut Pro Workflow Extensions aren't great for complicated/time consuming stuff like this either, because as soon as you close the Workflow Extension window, even if it's only accidentally because you've changed Final Cut Pro Workspaces, the Workflow Extension is "destroyed" - which is why the file list is reset each time you close the window.
